bin/ck8s

```diff
@@ -56,8 +56,10 @@ for arg in "$@"; do
   esac
 done
 
+export CK8S_CLUSTER="${2}"
```

Comment:
This won't be correct for commands that takes something other than a cluster as the second argument. For example `ck8s ops kubectl ...`.

Reply:
Yes, it would in that case be set something like this to override the variable:
```
    ops)
        CK8S_CLUSTER="${3}"
        case "${2}" in
```
But like I said, none of the other commands support using this global environment variable atm, so this would need more work either in this issue or in another issue

Reply:
Otherwise you could export it for each command that support it like I initially did in this commit: https://github.com/elastisys/compliantkubernetes-apps/pull/1845/commits/593ef62aaa21f7a21ec0c0989cf4e37e13451a7c

Reply:
Again, I would have loved that it was simple a variable you set in your shell which defaulted to `both` instead of an argument but that's for the future I guess... :smile:

Reply:
Why are we using a global variable instead of passing it to functions/scripts?
To me it feels a bit dangerous expecting global variables to be set.

Reply:
> Why are we using a global variable instead of passing it to functions/scripts? To me it feels a bit dangerous expecting global variables to be set.

How so? The risk of accidentally writing the wrong letters when you set the variable vs writing the argument seems equally as dangerous. Similarly forgetting to set the variable to the right cluster or to rewrite your copy-pasted line seems equally as dangerous as well.

In this case neither is a problem though because the command-line isn't changed, only the variable passing.

Reply:
If I had rewritten this from scratch I would have made it configurable both using command-line flags and environment variables. Not positional arguments.

Reply:
> Why are we using a global variable instead of passing it to functions/scripts? To me it feels a bit dangerous expecting global variables to be set.

Scripts that sources the `scripts/migration/lib.sh` now needs a global environment variable `CK8S_CLUSTER` to be able to validate configuration files etc per cluster. I changed it this way in a [previous PR](https://github.com/elastisys/compliantkubernetes-apps/commit/6cd8a4ee0692703df251dbc23bce804e00126618#diff-d157d0f7a32fe5a99591a42778e449309300204efe637caef726eefdceb9f0de) as a means to allow _per cluster_ commands, as I did not see any other good approach for solving this.

Reply:
@simonklb I updated the PR, can I mark this as solved?